src/namer.h

```diff
@@ -0,0 +1,129 @@
+#ifndef FLATBUFFERS_NAMER
+#define FLATBUFFERS_NAMER
+
+#include "flatbuffers/idl.h"
+#include "flatbuffers/util.h"
+
+namespace flatbuffers {
+
+// `Namer` applies style configuration to symbols in generated code. It manages
+// casing, escapes keywords, and object API naming.
+// TODO: Refactor all code generators to use this.
+class Namer {
+ public:
+  struct Config {
+    // Symbols in code.
+    Case types;
+    Case constants;
+    Case methods;
+    Case functions;
+    Case fields;
+    Case variants;
+
+    // Namespaces
+    Case namespaces;
+    std::string namespace_seperator;
+
+    // Object API.
+    std::string object_prefix;
+    std::string object_suffix;
+
+    // Keywords.
+    std::string keyword_prefix;
+    std::string keyword_suffix;
+
+    // Files.
+    Case filenames;
+    Case directories;
+    std::string output_path;
+    std::string filename_suffix;
+    std::string filename_extension;
+
+    // DO NOT SUBMIT
+    // comment/doc comment.
+  };
+  Namer(Config config, std::set<std::string> keywords)
+      : config_(config), keywords_(std::move(keywords)) {}
+
+  // Returns a mutable reference to the config so it can be modified **before**
+  // code generation. The intent is for this class to be const while generating
+  // code.
+  Config &GetConfig() { return config_; }
```

Comment:
Do we have a use case for this? If not, i wouldn't add till needed.

Reply:
I see, you have it below for modifying it. I added comments there, we should try to make this const always.

Reply:
Removed